src/modules/keyboardmanager/common/KeyDelay.cpp

```diff
@@ -0,0 +1,167 @@
+#include "pch.h"
+#include "KeyDelay.h"
+
+#define LONG_PRESS_DELAY_MILLIS 900
+#define ON_HOLD_WAIT_TIMEOUT_MILLIS 50
+
+KeyDelay::~KeyDelay()
+{
+    _quit = true;
+    _cv.notify_all();
+    _delayThread.join();
+}
+
+void KeyDelay::KeyEvent(LowlevelKeyboardEvent* ev)
+{
+    std::lock_guard guard(_queueMutex);
+    _queue.push({ ev->lParam->time, ev->wParam });
+    _cv.notify_all();
+}
+
+KeyTimedEvent KeyDelay::NextEvent()
+{
+    auto ev = _queue.front();
+    _queue.pop();
+    return ev;
+}
+
+bool KeyDelay::CheckIfMillisHaveElapsed(DWORD first, DWORD last, DWORD duration)
+{
+    if (first < last && first <= first + duration) 
+    {
+        return first + duration < last;
+    }
+    else
+    {
+        first += ULONG_MAX / 2;
+        last += ULONG_MAX / 2;
+        return first + duration < last;
+    }

```

Comment:
Is this to handle overflow rather than just using `first < last`?

Reply:
Yes, exactly.